jax_galsim/wcs.py

```diff
@@ -0,0 +1,922 @@
+        header["GS_SCALE"] = (self.scale, "GalSim image scale")
+        return self.affine()._writeLinearWCS(header, bounds)
+
+    @staticmethod
+    def _readHeader(header):
+        scale = header["GS_SCALE"]
+        return PixelScale(scale)
+
+    def _newOrigin(self, origin, world_origin):
+        return OffsetWCS(self._scale, origin, world_origin)
+
+    def copy(self):
+        return PixelScale(self._scale)
+
+    def __eq__(self, other):
+        return self is other or (
+            isinstance(other, PixelScale) and self.scale == other.scale
+        )
+
+    def __repr__(self):
+        return "galsim.PixelScale(%r)" % self.scale
+
+    def __hash__(self):
+        return hash(repr(self))
+
+
+@_wraps(_galsim.JacobianWCS)
+@register_pytree_node_class
+class JacobianWCS(LocalWCS):
+    def __init__(self, dudx, dudy, dvdx, dvdy):
+        self._color = None
+        self._params = {"dudx": dudx, "dudy": dudy, "dvdx": dvdx, "dvdy": dvdy}
+        self._det = dudx * dvdy - dudy * dvdx
+
+    # Help make sure JacobianWCS is read-only.
+    @property
+    def dudx(self):
+        """du/dx"""
+        return self._params["dudx"]
+
+    @property
+    def dudy(self):
+        """du/dy"""
+        return self._params["dudy"]
+
+    @property
+    def dvdx(self):
+        """dv/dx"""
+        return self._params["dvdx"]
+
+    @property
+    def dvdy(self):
+        """dv/dy"""
+        return self._params["dvdy"]
+
+    def _u(self, x, y, color=None):
+        return self.dudx * x + self.dudy * y
+
+    def _v(self, x, y, color=None):
+        return self.dvdx * x + self.dvdy * y
+
+    def _x(self, u, v, color=None):
+        #  J = ( dudx  dudy )
+        #      ( dvdx  dvdy )
+        #  J^-1 = (1/det) (  dvdy  -dudy )
+        #                 ( -dvdx   dudx )
+        return (self.dvdy * u - self.dudy * v) / self._det
+
+    def _y(self, u, v, color=None):
+        return (-self.dvdx * u + self.dudx * v) / self._det
+
+    def _profileToWorld(self, image_profile, flux_ratio, offset):
+        return _Transform(
+            image_profile,
+            (self.dudx, self.dudy, self.dvdx, self.dvdy),
+            flux_ratio=flux_ratio / self._pixelArea(),
+            offset=offset,
+        )
+
+    def _profileToImage(self, world_profile, flux_ratio, offset):
+        return _Transform(
+            world_profile,
+            (
+                self.dvdy / self._det,
+                -self.dudy / self._det,
+                -self.dvdx / self._det,
+                self.dudx / self._det,
+            ),
+            flux_ratio=flux_ratio * self._pixelArea(),
+            offset=offset,
+        )
+
+    def _pixelArea(self):
+        return abs(self._det)
+
+    def getMatrix(self):
+        """Get the Jacobian as a NumPy matrix:
+
+        numpy.array( [[ dudx, dudy ],
+                      [ dvdx, dvdy ]] )
+        """
+        return jnp.array([[self.dudx, self.dudy], [self.dvdx, self.dvdy]], dtype=float)
+
```

Comment:
The function `getDecomposition` is missing here, probably bc we don't need it yet, but we should remember it somehow